numpy/testing/_private/utils.py

```diff
@@ -706,6 +711,16 @@ def chk_same_position(x_id, y_id, hasval='nan'):
                                 % (hasval), verbose=verbose, header=header,
                                 names=('x', 'y'), precision=precision)
             raise AssertionError(msg)
+        return x_id if x_id.shape else y_id
+
+    def remove_flagged(x, y, flag):
+        if flag.all():
+            return array(()), array(())
+        if x.shape:
+            x = x[~flag]
+        if y.shape:
+            y = y[~flag]
+        return x, y
```

Comment:
I don't understand this function - can't it just be written `return x[~flag], y[~flag]`?

Reply:
No, `x` or `y` can be a scalar while the other is an array. This is why the flags were treated separately before.

Reply:
The issue here is avoiding `arr_0d[bool_nd]`, right? Would `x, y = np.broadcast_arrays(x, y)` be off the table?

Reply:
Not crazy, but I'd rather stick with what I have here...